### PR TITLE
assimp, new recipe, wip

### PR DESCRIPTION
**Pull request by Begasus**
Needs to be checked on x86_64

### CODE REVIEW

**Comment by miqlas** [comment]
Closes #1645

File: media-libs/assimp/assimp-4.0.1.recipe [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+	cmake -DASSIMP_BIN_INSTALL_DIR=$binDir \
+		-DASSIMP_LIB_INSTALL_DIR=$libDir \
+		-DASSIMP_INCLUDE_INSTALL_DIR=$includeDir ..
```
Comment (korli) [inline media-libs/assimp/assimp-4.0.1.recipe]: These INSTALL_DIR could theorically be replaced by $cmakeDirArgs, provided that https://github.com/assimp/assimp/blob/master/CMakeLists.txt#L186 is extended: if( UNIX or HAIKU )

File: media-libs/assimp/assimp-4.0.1.recipe [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+REQUIRES_devel="
+	assimp$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libGL$secondaryArchSuffix
+	devel:libglu$secondaryArchSuffix
+	devel:libminizip$secondaryArchSuffix
+	devel:libIL$secondaryArchSuffix
```
Comment (korli) [inline media-libs/assimp/assimp-4.0.1.recipe]: libIL is a dependency for assimp_qt_viewer. This shouldn't be needed.

Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: removed DevIL from the recipe ...

File: media-libs/assimp/assimp-4.0.1.recipe (hunk 1 above)
Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: Have to look into this how to use it ;)

Comment (korli) [reply media-libs/assimp/assimp-4.0.1.recipe]: That would be: cmake $cmakeDirArgs ..

provided that https://github.com/assimp/assimp/blob/master/CMakeLists.txt#L186 is extended: if( UNIX or HAIKU )

Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: I get the part of " cmake $cmakeDirArgs .. ", just not sure how to extend the correct way of extending ( UNIX or HAIKU ) ...

Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: PS, not too familiar with cmake myself ...

Comment (miqlas) [reply media-libs/assimp/assimp-4.0.1.recipe]: @korli: arent't we UNIX for cmake already?

Comment (korli) [reply media-libs/assimp/assimp-4.0.1.recipe]: Miqlas right, wasn't aware of it. Then it should just work.

Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: not using the INSTALL_DIR options fails on x86_gcc2 for secondaryArchSuffix ...

Comment (korli) [reply media-libs/assimp/assimp-4.0.1.recipe]: Begasus, fine, the packages don't always correctly use the first provided by cmakeDirArgs. Ok to merge?

Comment (Begasus) [reply media-libs/assimp/assimp-4.0.1.recipe]: Aside from removing DevIL from the recipe it builds ok here, miqlas tested the build on x86_64 :)

**Pull request merged**